gokart/gcs_obj_metadata_client.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import copy
+from logging import getLogger
+from typing import Any, Dict, Optional, Union
+from urllib.parse import urlsplit
+
+import luigi
+from googleapiclient.model import makepatch
+
+from gokart.gcs_config import GCSConfig
+
+logger = getLogger(__name__)
+
+
+class GCSObjectMetadataClient(object):
+    """
+    This class is Utility-Class, so should not be initialized.
+    This class used for adding metadata as labels.
+    """
+
+    # This is the copied method of luigi.gcs._path_to_bucket_and_key(path).
+    @staticmethod
+    def path_to_bucket_and_key(path):
```

Comment:
please add type annotations, though this function is copied.
```suggestion
    def path_to_bucket_and_key(path: str) -> tuple[str, str]:
```